In [2]:
from pathlib import Path

data_folder = Path("../Dataset_rev")
file = data_folder / "videos_rev2.csv"

In [4]:
import pandas as pd

df = pd.read_csv(file)
df['Month']= pd.to_datetime(df['user_created_at']).dt.month
df['numWatched'] = df['numWatched'].fillna(0)
df

id  \
0     6e884071-8c0e-4782-aa40-e0fd78243360   
1     ac90ff16-24b4-46bc-ae72-2da847f9e6c9   
2     fb5eaf0d-e72c-4523-b5c5-4d9c323cef5f   
3     bbf89dcb-7fbd-439d-9a7d-36ef7a1c2aa3   
4     4c613ec6-fa6f-47e8-b4ab-658e2f6a644d   
...                                    ...   
4995  f6fec2d4-77d6-46ea-b41c-d97154bd5238   
4996  f90ef4f1-b5ab-431e-9fc8-cf7359138168   
4997  ad00eac6-19d5-48f9-bed7-066b77546191   
4998  1100779d-e85f-4836-a154-3b4d543f910c   
4999  9708288c-b868-4d01-9d81-d4b2aa0de750   

                              attribute_id  \
0     4712e9e6-bef1-4646-b21e-ca5336901bf9   
1     4f91d2b6-bf42-4d3f-a8e8-acc3b07f647e   
2     80f5b187-f6e6-40f2-9b6e-fe62b67049a8   
3     02b2b662-1b39-4d91-9417-dddc9451d8e1   
4     06c496ee-e884-42f9-ba39-010a66ed5296   
...                                    ...   
4995  e339ba35-ffef-403b-a17c-dd218fdbb915   
4996  06c496ee-e884-42f9-ba39-010a66ed5296   
4997  4f91d2b6-bf42-4d3f-a8e8-acc3b07f647e   
4998  25e7fab9-abfb-4a74-a8cb-c3adbc6e3531   
4999  4712e9e6-bef1-4646-b21e-ca5336901bf9   

                                  music_id  \
0     4ee0de5b-e27f-4d73-a3c6-52cf26e36c54   
1     4ee0de5b-e27f-4d73-a3c6-52cf26e36c54   
2     d53234f4-1efb-4299-97c1-11658118cba5   
3     19e4db13-d854-4913-a8ca-3612ee1a6071   
4     b5b679a5-e769-4d95-82a6-f75e439b1ef9   
...                                    ...   
4995  19e4db13-d854-4913-a8ca-3612ee1a6071   
4996  8ff50feb-3b5d-4dd0-84b0-8409753bd166   
4997  e2bf5d12-edc2-47cb-b23f-1f4f7c1a2292   
4998  da2fc9b0-32d5-4a1b-8b40-2706eb133bf0   
4999  19e4db13-d854-4913-a8ca-3612ee1a6071   

                              animation_id  duration  \
0     17dab81c-6180-4a4b-b539-91978a726cfd      5.34   
1     6d7a0545-0bca-4bf4-a306-931e57d8ffe2      3.88   
2     716c02df-bd3a-41c9-b2be-86fb74ff3255      5.15   
3     b7bd5204-2976-4eb0-ab71-508a848bc325      6.50   
4     17dab81c-6180-4a4b-b539-91978a726cfd      4.38   
...                                    ...       ...   
4995  ff1a548b-e307-474c-a405-6370ca1a0639      3.82   
4996  08dda558-a151-45a2-8e43-1ba6f2891800      5.64   
4997  ff1a548b-e307-474c-a405-6370ca1a0639      4.72   
4998  ace4c5e2-ef86-429a-aad3-23bef7a1c9f1      6.44   
4999  b7bd5204-2976-4eb0-ab71-508a848bc325      2.72   

                                   creator           created_at  num_shares  \
0     73ec1ac2-d148-45ac-94a8-9bdff51730d7  2021-08-04 10:42:39          12   
1     0230f9af-e89f-4886-8322-52a38eaf991c  2021-11-27 10:06:43           3   
2     fbf93fbf-a935-436f-8516-8acb863e779f  2021-02-26 04:18:53           4   
3     f7a2adb8-44d7-4408-a2a9-3a593fc6d626  2021-11-27 02:08:02           9   
4     d9f9538f-1b7e-4f35-805d-924b309d75d8  2021-11-17 11:06:32          14   
...                                    ...                  ...         ...   
4995  4db64146-3864-40d1-9908-faa3f4a04a80  2021-11-08 04:21:06           2   
4996  4bcc7557-3760-49e3-895a-f122d9d112f1  2021-09-11 23:29:23          18   
4997  0ef242cf-8c59-45d7-acfc-dadce146c907  2021-11-22 16:46:18           4   
4998  9b8274ec-0e8f-47fb-8b16-07e1ec115798  2021-07-09 18:02:48           8   
4999  7884bd3e-d48f-4afd-87dd-2a7552c04480  2021-10-13 06:47:46          19   

                                   game_id  is_ar      user_created_at  \
0     8bd4c7be-fe87-45ad-990d-d131cd3d9c77   True  2021-07-01 18:27:53   
1     6ca0f645-a16b-42b0-9dc2-2ae952c83591   True  2021-06-04 20:14:39   
2     6ca0f645-a16b-42b0-9dc2-2ae952c83591   True  2021-02-04 04:14:46   
3     8bd4c7be-fe87-45ad-990d-d131cd3d9c77  False  2021-11-20 16:11:07   
4     2609b6b5-3a08-451e-bce5-75fe506559d2  False  2021-09-18 15:58:48   
...                                    ...    ...                  ...   
4995  6ca0f645-a16b-42b0-9dc2-2ae952c83591  False  2021-08-27 19:33:54   
4996  66653b7c-b600-49b3-a58c-d1862e91d426  False  2021-08-05 21:42:46   
4997  66653b7c-b600-49b3-a58c-d1862e91d426  False  2021-11-09 21:20:32   
4998  2609b6b5-3a08

In [5]:
# correlation
df_corr = df.drop(columns = ['id','user_latitude', 'user_longitude']).apply(lambda x: x.factorize()[0]).corr()
df_corr

attribute_id  music_id  animation_id  duration   creator  \
attribute_id         1.000000 -0.015368      0.044513 -0.017300  0.020050   
music_id            -0.015368  1.000000      0.029106 -0.006031 -0.009107   
animation_id         0.044513  0.029106      1.000000  0.018153  0.023282   
duration            -0.017300 -0.006031      0.018153  1.000000  0.100899   
creator              0.020050 -0.009107      0.023282  0.100899  1.000000   
created_at          -0.010935 -0.008996      0.000139  0.131176  0.245985   
num_shares          -0.009701 -0.002394      0.011076  0.004079  0.005491   
game_id              0.011190 -0.005624      0.006785 -0.010920  0.006379   
is_ar               -0.007488  0.008321     -0.000285 -0.010385 -0.014521   
user_created_at      0.020050 -0.009107      0.023282  0.100899  1.000000   
class               -0.014113 -0.008554     -0.000833 -0.003465  0.017209   
user_city            0.006247  0.009338      0.015438  0.024636  0.259239   
user_state           0.019394  0.006984      0.005927  0.043562  0.412482   
user_country         0.018733  0.010769      0.022323  0.034981  0.116939   
numWatched          -0.017964  0.018808      0.001197  0.009823 -0.002214   
Month               -0.010185  0.002964     -0.018915  0.018885  0.029136   

                 created_at  num_shares   game_id     is_ar  user_created_at  \
attribute_id      -0.010935   -0.009701  0.011190 -0.007488         0.020050   
music_id          -0.008996   -0.002394 -0.005624  0.008321        -0.009107   
animation_id       0.000139    0.011076  0.006785 -0.000285         0.023282   
duration           0.131176    0.004079 -0.010920 -0.010385         0.100899   
creator            0.245985    0.005491  0.006379 -0.014521         1.000000   
created_at         1.000000    0.007511 -0.016321 -0.002903         0.245985   
num_shares         0.007511    1.000000  0.004043 -0.004974         0.005491   
game_id           -0.016321    0.004043  1.000000  0.003472         0.006379   
is_ar             -0.002903   -0.004974  0.003472  1.000000        -0.014521   
user_created_at    0.245985    0.005491  0.006379 -0.014521         1.000000   
class             -0.003718   -0.031157  0.049873 -0.006537         0.017209   
user_city          0.052842    0.012132  0.001851  0.013749         0.259239   
user_state         0.081871   -0.010655  0.026375  0.008148         0.412482   
user_country       0.010932   -0.010568 -0.006129  0.012576         0.116939   
numWatched        -0.015276    0.027529  0.012759 -0.005446        -0.002214   
Month             -0.001966    0.005749 -0.000526 -0.005330         0.029136   

                    class  user_city  user_state  user_country  numWatched  \
attribute_id    -0.014113   0.006247    0.019394      0.018733   -0.017964   
music_id        -0.008554   0.009338    0.006984      0.010769    0.018808   
animation_id    -0.000833   0.015438    0.005927      0.022323    0.001197   
duration        -0.003465   0.024636    0.043562      0.034981    0.009823   
creator          0.017209   0.259239    0.412482      0.116939   -0.002214   
created_at      -0.003718   0.052842    0.081871      0.010932   -0.015276   
num_shares      -0.031157   0.012132   -0.010655     -0.010568    0.027529   
game_id          0.049873   0.001851    0.026375     -0.006129    0.012759   
is_ar           -0.006537   0.013749    0.008148      0.012576   -0.005446   
user_created_at  0.017209   0.259239    0.412482      0.116939   -0.002214   
class            1.000000   0.000371    0.023287      0.043239    0.000400   
user_city        0.000371   1.000000    0.247551      0.060543    0.029576   
user_state       0.023287   0.247551    1.000000      0.318691   -0.003528   
user_country     0.043239   0.060543    0.318691      1.000000   -0.006871   
numWatched       0.000400   0.029576   -0.003528     -0.006871    1.000000   
Month           -0.059557   0.037888    0.020944      0.042830   -0.001899   

                   

In [6]:
corr_set = set()
for col in df_corr:
    for index in df_corr.index:
        if df_corr.loc[index].loc[col] >= 0.5 and index != col:
            corr_set.add(tuple(sorted((index, col))))

In [7]:
corr_set

{('creator', 'user_created_at')}

### no meaningful correlation in user file
The number of times a video was shared (num_shares) and the number of watched videos (numWatched) has only 0.027529	 correlation. 

In [4]:
def getCount_col (col):
    cnt_df = df.groupby(df[col]).size().reset_index(name='counts')
    cnt_df.set_index(col, inplace=True)
    cnt_df.sort_values(by=['counts'], ascending=False, inplace=True)
    
    return cnt_df

In [5]:
getCount_col ('class')

counts
class                
Gamer             213
Super Gamer       209
Super Creator     206
Creator           186
Viewer            186

In [8]:
df.sort_values(by=['numWatched'], ascending=False)

id  \
3778  241ed4cb-e1bc-4892-9d29-304a9b5f8848   
831   103b0a8e-6e31-4b89-8fd4-bc0c428744c2   
2046  5cc23c88-c8a9-46fa-bbf1-0e687bcde7d4   
995   53bafd7f-832f-403d-b971-07c1ecdeec39   
3529  4950c21e-2739-4daa-bd66-8a8a81a693c1   
...                                    ...   
1574  19383191-1d30-40a0-9387-7231e21725fb   
2341  8ce22b7d-394d-4e64-8e31-9ae55173a632   
3016  8e23162d-02ca-4e28-b289-050880ac70bd   
1945  6fe34e47-095c-43ae-a2e2-5917aa00dc6b   
2269  96834ffa-2fee-4018-a583-7b570223cc99   

                              attribute_id  \
3778  7ef48c9e-776d-4d24-8c2b-e3675b295767   
831   80f5b187-f6e6-40f2-9b6e-fe62b67049a8   
2046  3390e253-7815-4a3c-8169-fd2ab532e166   
995   4712e9e6-bef1-4646-b21e-ca5336901bf9   
3529  7ef48c9e-776d-4d24-8c2b-e3675b295767   
...                                    ...   
1574  02b2b662-1b39-4d91-9417-dddc9451d8e1   
2341  3390e253-7815-4a3c-8169-fd2ab532e166   
3016  e339ba35-ffef-403b-a17c-dd218fdbb915   
1945  25e7fab9-abfb-4a74-a8cb-c3adbc6e3531   
2269  4f91d2b6-bf42-4d3f-a8e8-acc3b07f647e   

                                  music_id  \
3778  c3721b09-6896-465a-8e13-4d73efd06874   
831   19e4db13-d854-4913-a8ca-3612ee1a6071   
2046  e2bf5d12-edc2-47cb-b23f-1f4f7c1a2292   
995   b5b679a5-e769-4d95-82a6-f75e439b1ef9   
3529  19e4db13-d854-4913-a8ca-3612ee1a6071   
...                                    ...   
1574  3de11516-8627-47c3-98ab-265b429cac91   
2341  8ff50feb-3b5d-4dd0-84b0-8409753bd166   
3016  4ee0de5b-e27f-4d73-a3c6-52cf26e36c54   
1945  da2fc9b0-32d5-4a1b-8b40-2706eb133bf0   
2269  b5b679a5-e769-4d95-82a6-f75e439b1ef9   

                              animation_id  duration  \
3778  b7bd5204-2976-4eb0-ab71-508a848bc325      3.40   
831   17dab81c-6180-4a4b-b539-91978a726cfd      4.79   
2046  b7bd5204-2976-4eb0-ab71-508a848bc325      5.02   
995   6d7a0545-0bca-4bf4-a306-931e57d8ffe2      2.12   
3529  ab5dc9f1-deb4-44d4-88ff-62e473a060f4      4.76   
...                                    ...       ...   
1574  4cfc4892-2c7e-46e3-8399-5b138285d394      5.70   
2341  542d7532-e88a-48ca-826f-c11f0f615bc4      3.18   
3016  435320a3-8d3d-4a9d-b76b-2f2e268f7619      4.34   
1945  6d7a0545-0bca-4bf4-a306-931e57d8ffe2      2.37   
2269  6d7a0545-0bca-4bf4-a306-931e57d8ffe2      4.81   

                                   creator           created_at  num_shares  \
3778  dd947420-df8a-4815-a81b-87931d33e108  2021-10-06 05:51:37          19   
831   513c26ab-70a1-4be4-97b3-837f62089c54  2021-12-10 05:18:57           1   
2046  495f4ac5-c45d-4e6c-bf21-5193239e9843  2021-04-15 19:33:41          11   
995   7ee05cfe-f40f-4e88-a637-7d6b3474b963  2021-10-23 04:16:49           1   
3529  de3c4db2-06b7-479d-bb62-5b39fe40dca7  2021-07-30 19:16:39          15   
...                                    ...                  ...         ...   
1574  c1c68262-ff75-42a5-bd04-398c43e9b0e1  2021-10-28 21:29:57          17   
2341  c7feb614-4f4e-44a0-a1e3-3dee86b5700f  2021-03-09 17:10:29           6   
3016  addab700-7a42-45e2-af49-219578b3c3ba  2021-08-28 15:37:47          11   
1945  8dc915ee-269c-4e0f-878a-c288c331f513  2021-06-16 03:36:14           3   
2269  5fd34ddd-4b07-4f11-a022-b65dd1c9080e  2021-09-15 20:47:23          15   

                                   game_id  is_ar      user_created_at  \
3778  66653b7c-b600-49b3-a58c-d1862e91d426   True  2021-01-21 07:48:00   
831   66653b7c-b600-49b3-a58c-d1862e91d426   True  2021-01-02 11:28:59   
2046  6ca0f645-a16b-42b0-9dc2-2ae952c83591  False  2021-01-19 16:53:37   
995   8bd4c7be-fe87-45ad-990d-d131cd3d9c77   True  2021-08-13 23:15:43   
3529  6ca0f645-a16b-42b0-9dc2-2ae952c83591  False  2021-04-24 12:01:54   
...                                    ...    ...                  ...   
1574  4d54f6ab-cd8f-43a5-9be0-c497c2c32d24   True  2021-10-11 21:22:00   
2341  2609b6b5-3a08-451e-bce5-75fe506559d2   True  2021-02-19 16:57:19   
3016  4d54f6ab-cd8f-43a5-9be0-c497c2c32d24  False  2021-05-03 15:46:53   
1945  6ca0f645-a16b

In [7]:
df.sort_values(by=['numWatched'], ascending=False)

id      user_created_at          class  \
48   39acf63e-c53c-43b9-a250-7a391070861a  2021-10-10 23:36:52        Creator   
826  3a4ff1db-8a58-4a3b-872d-8bf143067225  2021-05-16 17:37:46         Viewer   
665  ffaed591-9da5-4094-9e93-040f4300cf29  2021-04-24 04:33:56    Super Gamer   
361  bd474dc1-7576-41ee-b18a-b2626b750a05  2021-08-30 05:47:07  Super Creator   
496  2d169120-94c0-4922-8842-e70abe8a0f0f  2021-05-21 02:30:51    Super Gamer   
..                                    ...                  ...            ...   
577  1b4d93d7-4d04-484c-953d-25b0c9cf5644  2021-10-19 09:04:34         Viewer   
685  a3d2cffc-8286-47ac-bd57-6d291b60248f  2021-01-11 02:39:11    Super Gamer   
413  4b304bba-650c-4249-85a0-c37b16637a1a  2021-04-01 10:35:33        Creator   
593  ff4f151f-6a94-4f02-980d-5056ff9cd449  2021-03-09 11:38:55         Viewer   
294  bf72d5aa-a63d-41c9-b395-0166b8315a29  2021-04-26 03:14:14         Viewer   

     user_latitude  user_longitude user_city         user_state  \
48         4.88441       101.96857       NaN           Kelantan   
826       33.95015      -118.03917       NaN         California   
665        4.73245       -74.26419       NaN                NaN   
361       12.18441        18.69303     Mongo              Guéra   
496       39.22493        42.85693       NaN                NaN   
..             ...             ...       ...                ...   
577       -1.60000       103.61667       NaN              Jambi   
685       48.35149         8.96317       NaN  Baden-Württemberg   
413       10.10649        76.35484       NaN             Kerala   
593       34.15429         3.50309   Messaad             Djelfa   
294      -19.72806       -50.19556       NaN       Minas Gerais   

      user_country  # of videos  numWatched  Month  
48        Malaysia          3.0          72     10  
826  United States          2.0          71      5  
665       Colombia          6.0          69      4  
361           Chad          3.0          68      8  
496         Turkey          3.0          68      5  
..             ...          ...         ...    ...  
577      Indonesia          4.0          32     10  
685        Germany          6.0          32      1  
413          India          5.0          32      4  
593        Algeria          6.0          32      3  
294         Brazil          6.0          30      4  

[1000 rows x 11 columns]